src/pipeline/world/collision_object.rs

```diff
@@ -174,6 +179,12 @@ impl<N: Real, T> CollisionObject<N, T> {
         self.query_type
     }
 
+    /// Sets the `GeometricQueryType` of the collision object.
+    #[inline]
+    pub fn set_query_type(&mut self, query_type: GeometricQueryType<N>) {
```

Comment:
This can't be a public method here because a change of `query_type` will break the collision world.
This should be a method of the `CollisionWorld` instead, which calls the following after changing the query type of a collision object `co`:

```rust
self.broad_phase.deferred_recompute_all_proximities_with(co.proxy_handle());
```

This is needed because the algorithms used for contact manifold computation are not the same as the ones for proximity detection.